src/Cardano/Wallet/Api/Types.hs

```diff
@@ -378,10 +375,11 @@ walletStateOptions = taggedSumTypeOptions $ TaggedObjectOptions
     , _contentsFieldName = "progress"
     }
 
-instance FromJSON TargetOutput where
-    parseJSON = genericParseJSON defaultRecordTypeOptions
-instance ToJSON TargetOutput where
-    toJSON = genericToJSON defaultRecordTypeOptions
+instance FromJSON (ApiT Coin) where
+    parseJSON = parseJSON @(Quantity "lovelace" Word64) >=>
```

Comment:
@KtorZ how do you like this for handling json instances? Is it too hacky?

Reply:
Not too hacky no :) Seems fine to me, though for consistency, I'd still go for `ApiT TxOut` in the above type, and, I think `parseJSON` could just be:

```hs
parseJSON = ApiT . Coin . getQuantity <$> parseJSON @(Quantity "lovelace" Word64)
```

I don't see the need to a monadic approach here. But that's neat-picking.

Reply:
```
parseJSON = ApiT . Coin . getQuantity <$> parseJSON @(Quantity "lovelace" Word64)
```

yes that looks nicer 

> I'd still go for ApiT TxOut in the above type
:+1: forgot to add that

Reply:
Could do

```haskell
{-# LANGUAGE StandaloneDeriving #-}
{-# LANGUAGE DerivingVia #-}

deriving via (Quantity "lovelace" Word64)
    instance ToJSON (ApiT Coin)

deriving via (Quantity "lovelace" Word64)
    instance FromJSON (ApiT Coin)
```
Sadly that breaks stylish-haskell, unless it is rebuilt with `haskell-src-exts-1.21.0` (tried yesterday)

Reply:
hm - never saw such sintax before

Reply:
It's new in 8.6 ✨ 
https://www.kosmikus.org/DerivingVia/deriving-via-paper.pdf
https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0023-deriving-via.rst

Reply:
I guess we are going to skip this for now until stylish-haskell on stackage lands with this option